dynamic_programming/minimum_edit_distance.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) using Dynamic Programing
+ *
+ * @details
+ *
+ * Given two strings str1 & str2 and we have to calculate the minimum
+ * number of operations (Insert, Remove, Replace) required to convert
+ * str1 to str2.
+ *
+ * ### Algorithm
+ *
+ * We will solve this problem using Naive recursion. But as we are
+ * approaching with a DP solution. So, we will take a DP array to
+ * store the solution of all sub-problems so that we don't have to
+ * perform recursion again and again. Now to solve the problem, We
+ * can traverse all characters from either right side of the strings
+ * or left side. Suppose we will do it from the right side. So, there
+ * are two possibilities for every pair of characters being traversed.
+ * 1. If the last characters of two strings are the same, Ignore
+ * the characters and get the count for the remaining string.
+ * So, we get the solution for lengths m-1 and n-1 in a DP array.
+ *
+ * 2. Else, (If last characters are not the same), we will consider all
+ * three operations (Insert, Remove, Replace) on the last character of
+ * the first string and compute the minimum cost for all three operations
+ * and take the minimum of three values in the DP array.
+ * For Insert: Recur for m and n-1
+ * For Remove: Recur for for m-1 and n
+ * For Replace: Recur for for m-1 and n-1
+ *
+ * @author [Nirjas Jakilim](github.com/nirzak)
+ */
+
+#include <cassert>     /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+
+namespace dynamic_programming {
+
+/**
+ * @namespace Minimum Edit Distance
+ * @brief Implementation of [Minimum Edit Distance](https://en.wikipedia.org/wiki/Edit_distance) algorithm
+ */
+
+namespace minimum_edit_distance {
+
+/**
+ * @brief Takes input of the cost of
+ * three operations: Insert, Replace and Delete
+ * and return the minimum cost among them.
+ * @params x for Insert, y for Replace and Z for Delet

```

Comment:
`x` is used to insert what? Complete the required information.
```suggestion
 * @param x used to insert <complete the required information>
 * @param y used to replace <complete the required information>
 * @param z used to delete <complete the required information>
```

Reply:
There's going to be someone that will need this information. We require your algorithm to be documented, and understandable, with refined code using C++11. I'm not going to use this algorithm, but someone else may see this algorithm, and that's why we add documentation here. 🙂

Reply:
Yeah and I am trying so. there's so many algorithms with even more bad documentation in this repository they are even published. I am trying to elaborate the whole program as much as possible but if you don't even read the first algorithm details then how can you know even the main concept. The main concept is to find the minimum number of insert, replace and delete operations required to change from one string to another. and you are asking to insert what?inser here not a verb here insert means the insert operation